design/BotKube POC for Namespace.md

```diff
@@ -0,0 +1,144 @@
+    enabled: false
+    channel: 'SLACK_CHANNEL'
+    token: 'SLACK_API_TOKEN'
+    notiftype: short  
+```
+### proposed stuucture 
+key concept: comm_config.yml contains a mapping of specific profile instead of specific channels 
+##### comm_config.yaml
+```
+  
+# Channels configuration
+communications:
+  # Settings for Slack
+  slack:
+    enabled: false
+    profile: 'PROFILE_NAME'   # listed under access_conf.yml to limit the level of access from different channels
+    token: 'SLACK_API_TOKEN'
+    notiftype: short  
+```
+
+##### access_conf.yml
+```
+profiles:
+  # based on use-case like profile for development environment
+  - name: 'DEVELOPMENT'     
+    channels:
+    - name: 'DEVELOPER_SLACK_CHANNEL'
+      namespaces:
+      - ns1
+      - ns2
+      kubectl:
+        enabled: true
+        commands:
+          # method which are allowed
+          verbs: ["get", "logs"]
+          # resource configuration which is allowed
+          resources: ["Deployments", "Pods", "Services"]
+          # enable notification about specific resources
+          configwatcher: false 
+          
+    - name: 'OPERATION_SLACK_CHANNEL'
+      namespaces:
+      - ns1
+      - ns2
+      kubectl:
+        enabled: true
+        commands:
+          # method which are allowed
+          verbs: ["get", "logs", "decsribe"]
+          # resource configuration which is allowed
+          resources: ["Deployments", "Pods", "Services"]
+          # enable notification about specific resources
+          configwatcher: true  
+   
+  # Profile: For Production resources 
+  - name: 'PRODUCTION'
+    channels:
+    - name: 'PRODUCTION_SLACK_CHANNEL'
+      namespaces:
+      - ns1
+      - ns2
+      kubectl:
+        enabled: true
+        commands:
+          # method which are allowed
+          verbs: ["get", "logs", "describe", "diff", "explain", "top"]
+          # resource configuration which is allowed
+          resources: ["Deployments", "Pods", "Services", "Nodes", "Ingresses", "Roles"]  
+          # enable notification about specific resources
+          configwatcher: true 
+          
+
+
+  # Profile: BOTKUBE_ADMIN 
+  - name: 'BOTKUBE_ADMIN'
+    channels:
+    - name: 'ADMIN_SLACK_CHANNEL'
+      namespaces:
+      # by default all namespaces will be included
+      # - ns1
+      # - ns2
+      kubectl:
+        enabled: true
+        commands:
+          # method which are allowed
+          verbs: ["get", "logs", "describe" ,"api-resources", "api-versions", "cluster-info", "diff", "explain", "top", "auth"]
+          # resource configuration which is allowed
+          resources: [ "Pods", "Services","Namespaces", "Nodes","ReplicationControllers", "PersistentVolumes", "PersistentVolumeClaims", "Secrets", "ConfigMaps", "Deployments", "DaemonSets", "ReplicaSets", "StatefulSets", "Ingresses", "Jobs", "Roles", "RoleBindings", "ClusterRoles"]  
+          # enable notification about specific resources
+          configwatcher: true 
```

Comment:
Wonder why configWatcher would be required and profile level ? It should be at comms level ?

Reply:
yes, I was also a bit confused to include it here or not but after thinking about some real-life scenario like a channel (say developer) having very specific and limited access should or should not be informed about the same specific resources as described in the profile. 
just as an additional option we can keep it as it is also.